### ISSUE TITLE
E3030 You must specify a valid value for EngineVersion (5.0.0). Valid values are ["3.6.0", "4.0", "4.0.0"]

### ISSUE THREAD

**Issue by leehagoodjames** [aws-cloudformation/cfn-lint#2806]
### CloudFormation Lint Version

cfn-lint 0.78.2

### What operating system are you using?

Mac

### Describe the bug

Running `sam validate --lint --template template.yml` yields:

```
E3030 You must specify a valid value for EngineVersion (5.0.0). Valid values are ["3.6.0", "4.0", "4.0.0"]
/Users/leejames/repos/bad-cards/template.yml:78:7
```

### Expected behavior

There would be no errors, since this engine is supported, as mentioned in https://docs.aws.amazon.com/documentdb/latest/developerguide/compatibility.html

### Reproduction template

```yaml
AWSTemplateFormatVersion: 2010-09-09
Transform:
  - AWS::CodeStar

Parameters:
  EnvironmentType:
    Type: String
    Default: beta
    AllowedValues:
      - beta
      - prod
  DBInstanceClass:
    Description: >-
      Instance class. Please refer to:
      https://docs.aws.amazon.com/documentdb/latest/developerguide/db-instance-classes.html#db-instance-classes-by-region
    Type: String
    Default: db.t3.medium
    AllowedValues:
      - db.t3.medium
    ConstraintDescription: >-
      Instance type must be of the ones supported for the region. Please refer
      to:
      https://docs.aws.amazon.com/documentdb/latest/developerguide/db-instance-classes.html#db-instance-classes-by-region
Resources:
  DocumentDBSecret:
    Type: AWS::SecretsManager::Secret
    Properties:
      Description: This is a Secrets Manager secret for the MongoDB Cluster
      GenerateSecretString:
        SecretStringTemplate: '{"username": "user"}'
        GenerateStringKey: password
        PasswordLength: 16
        ExcludeCharacters: "\"'@/\\"
  SecretDocumentDBAttachment:
    Type: AWS::SecretsManager::SecretTargetAttachment
    Properties:
      SecretId:
        Ref: DocumentDBSecret
      TargetId:
        Ref: DBCluster
      TargetType: AWS::DocDB::DBCluster
  DBCluster:
    Type: 'AWS::DocDB::DBCluster'
    DeletionPolicy: Delete
    Properties:
      DBClusterIdentifier: !Sub app-${EnvironmentType}-db
      MasterUsername: 
        Fn::Sub: "{{resolve:secretsmanager:${DocumentDBSecret}::username}}"
      MasterUserPassword: 
        Fn::Sub: "{{resolve:secretsmanager:${DocumentDBSecret}::password}}"
      EngineVersion: 5.0.0
  DBInstance:
    Type: 'AWS::DocDB::DBInstance'
    Properties:
      DBClusterIdentifier: !Ref DBCluster
      DBInstanceIdentifier: !Sub app-${EnvironmentType}-db-primary
      DBInstanceClass: !Ref DBInstanceClass
Outputs:
  ClusterId:
    Value: !Ref DBCluster
  ClusterEndpoint:
    Value: !GetAtt DBCluster.Endpoint
  ClusterPort:
    Value: !GetAtt DBCluster.Port
  EngineVersion:
    Value: 5.0.0
```

**Comment by kddejong** [comment]
Using cfn-lint v0.78.2
```
➜  cfn-python-lint git:(main) cfn-lint local/issue/2806.yaml
➜  cfn-python-lint git:(main) cfn-lint --version
cfn-lint 0.78.2
```

Fixed in pr #2800
